File: 算法/代码随想录算法PDF/00调试源代码/350.两个数组的交集-ii.cpp
/*
 * @lc app=leetcode.cn id=350 lang=cpp
 *
 * [350] 两个数组的交集 II
 */

// @lc code=start
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        sort(nums1.begin(), nums1.end());  // 给两个数组排序
        sort(nums2.begin(), nums2.end());
        int point1 = 0, point2 = 0; // 两个数组的指针
        while (point1 < nums1.size() && point2 < nums2.size()) {
            if (nums1[point1] < nums2[point2]) {
                point1++;
            } else if (nums1[point1] > nums2[point2]) {
                point2++;
            } else {
                res.push_back(nums1[point1]);
                point1++;
                point2++;
            }
        }
        return res;
    }
};
// @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/350.两个数组的交集-ii.py
#
# @lc app=leetcode.cn id=350 lang=python3
#
# [350] 两个数组的交集 II
#

# @lc code=start
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        nums1.sort()
        nums2.sort()
        point1, point2 = 0, 0
        while point1 < len(nums1) and point2 < len(nums2):
            if nums1[point1] < nums2[point2]:
                point1 += 1
            elif nums1[point1] > nums2[point2]:
                point2 += 1
            else:
                res.append(nums1[point1])
                point1 += 1
                point2 += 1
        return res
# @lc code=end

